storage/diskmetricstore_test.go

```diff
@@ -812,6 +873,27 @@ func checkMetricFamilies(dms *DiskMetricStore, expectedMFs ...*dto.MetricFamily)
 	return nil
 }
 
+func checkMetricFamilyGroups(dms *DiskMetricStore, expectedMFMap map[uint64]map[string]string) error {
+	mfMap := dms.GetMetricFamiliesMap()
+
+	if expected, got := len(expectedMFMap), len(mfMap); expected != got {
+		return fmt.Errorf("expected %d metric families in map, but got %d", expected, got)
+	}
+
+	for k, v := range mfMap {
+		if innerMap, ok := expectedMFMap[k]; ok {
+			for metricName, metricString := range innerMap {
```

Comment:
You should probably also assess that `len(innerMap) == len(v.Metrics)`. Otherwise, you would miss if `v.Metrics` contains more metrics than expected.

Reply:
True . Thanks for pointing that out :)

Reply:
This has been done.